assets/js/params/File/UploadedFileSelect.js

```diff
@@ -108,8 +110,10 @@ const UploadedFileSelect = React.memo(function UploadedFileSelect ({ value, file
 
   return (
     <>
-      <button className='uploaded-file-select' onClick={open} title={canOpen ? 'Choose a previously-uploaded file' : undefined} disabled={!canOpen}>
-        {`File ${valueIndex === -1 ? '[NONE]' : (valueIndex + 1)} of ${files.length}`}
+      <button className='uploaded-file-select' onClick={open} title={canOpen ? i18n._(t('js.params.File.UploadedFileSelect.choosePreviousFile.hoverText')`Choose a previously-uploaded file`) : undefined} disabled={!canOpen}>
+        <Trans id='js.params.File.UploadedFileSelect.choosePreviousFile.text' description='The first parameter is the number of the file (or js.params.File.UploadedFileSelect.choosePreviousFile.none) and the second is the total number of files'>
+            File {valueIndex === -1 ? <Trans id='js.params.File.UploadedFileSelect.choosePreviousFile.none'>[NONE]</Trans> : (valueIndex + 1)} of {files.length}
```

Comment:
Let's avoid this nested-`<Trans>` pattern. Can we make the translation string look something like `File {fileNumber} of {nFiles}`?

Reply:
Since the `{fileNumber}` may be "[NONE]", we can't avoid nesting. We can make it easier for the programmer to read, by splitting it to something like
```
  const nFiles = files.length
  const fileNumber = valueIndex === -1 ? <Trans id='js.params.File.UploadedFileSelect.choosePreviousFile.none'>[NONE]</Trans> : (valueIndex + 1)
        <Trans id='js.params.File.UploadedFileSelect.choosePreviousFile.text' description='The parameter {fileNumber} is the number of the file (or js.params.File.UploadedFileSelect.choosePreviousFile.none) and {nFiles} is the total number of files'>
            File {fileNumber} of {nFiles}
        </Trans>
```

Reply:
Yes, this is what I'm aiming for.